apps/web/src/components/layout/components/EnsureOnboardingComplete.tsx

```diff
@@ -1,16 +1,25 @@
 import { Navigate, useLocation } from 'react-router-dom';
 import { useAuth } from '../../../hooks/useAuth';
 import { ROUTES } from '../../../constants/routes';
-import { useBlueprint } from '../../../hooks/index';
+import { useBlueprint, useRedirectURL } from '../../../hooks';
 
 export function EnsureOnboardingComplete({ children }: any) {
   useBlueprint();
   const location = useLocation();
+  const { getRedirectURL } = useRedirectURL();
   const { currentOrganization, environmentId } = useAuth();
 
   if ((!currentOrganization || !environmentId) && location.pathname !== ROUTES.AUTH_APPLICATION) {
     return <Navigate to={ROUTES.AUTH_APPLICATION} replace />;
   }
 
+  const redirectURL = getRedirectURL();
+  if (redirectURL) {
+    // Note: Do not use react-router-dom. The version we have doesn't do instant cross origin redirects.
+    window.location.href = redirectURL;
```

Comment:
```suggestion
    window.location.replace(redirectURL);
```
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Assuming that we don't want the intermediate history if this is an auto-redirect‏